assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/convertors.py

```diff
@@ -0,0 +1,251 @@
+        :param pip_requirements: Path to pip requirements file
+        :type pip_requirements: Optional[str]
+        :param conda_env: Path to conda environment yaml file
+        :type conda_env: Optional[str]
+        :param code_path: A list of local filesystem paths to Python file dependencies
+        :type code_path: List[str]
+
+        """
+        signatures = self._signatures or self.get_model_signature()
+
+        mlflow.pyfunc.save_model(
+            path=self._output_dir,
+            python_model=mlflow_model_wrapper,
+            artifacts=artifacts_dict,
+            pip_requirements=pip_requirements,
+            conda_env=conda_env,
+            signature=signatures,
+            code_path=code_path,
+            metadata={"model_name": self._model_id},
+        )
+
+        logger.info("Model saved successfully.")
+
+    def _validate(self, translate_params):
+        """Validate translate parameters."""
+        if not translate_params.get("task"):
+            raise Exception("task is a required parameter for pyfunc flavor.")
+        task = translate_params["task"]
+        if not SupportedTasks.has_value(task):
+            raise Exception(f"Unsupported task {task} for pyfunc flavor.")
+
+
+class MMLabDetectionMLflowConvertor(PyFuncMLFLowConvertor):
+    """PyFunc MLfLow convertor for detection models from MMLab."""
+
+    MODEL_DIR = os.path.join(os.path.dirname(__file__), "vision")
+
+    def __init__(self, **kwargs):
+        """Initialize MLflow convertor for vision models."""
+        super().__init__(**kwargs)
+        if not MMLabDetectionTasks.has_value(self._task):
+            raise Exception("Unsupported vision task")
+
+    def get_model_signature(self) -> ModelSignature:
+        """Return MLflow model signature with input and output schema for the given input task.
+
+        :return: MLflow model signature.
+        :rtype: mlflow.models.signature.ModelSignature
+        """
+        input_schema = Schema(
+            [
+                ColSpec(VisionMLFlowSchemaLiterals.INPUT_COLUMN_IMAGE_DATA_TYPE,
+                        VisionMLFlowSchemaLiterals.INPUT_COLUMN_IMAGE)
+            ]
+        )
+
+        if self._task in [MMLabDetectionTasks.MM_OBJECT_DETECTION.value,
+                          MMLabDetectionTasks.MM_INSTANCE_SEGMENTATION.value]:
+            output_schema = Schema(
+                [
+                    ColSpec(VisionMLFlowSchemaLiterals.OUTPUT_COLUMN_DATA_TYPE,
+                            VisionMLFlowSchemaLiterals.OUTPUT_COLUMN_BOXES),
+                ]
+            )
+        else:
+            raise NotImplementedError(f"Task type: {self._task} is not supported yet.")
+        return ModelSignature(inputs=input_schema, outputs=output_schema)
+
+    def save_as_mlflow(self):
+        """Prepare model for save to MLflow."""
+        sys.path.append(self.MODEL_DIR)
+        from detection_predict import ImagesDetectionMLflowModelWrapper
```

Comment:
would this throw an import issue?

Reply:
No, this would not. We import it this way, so that the model wrapper is pickled appropriately.